apps/src/applab/applab.js

```diff
@@ -367,12 +367,24 @@ function shouldRenderFooter() {
   return studioApp.share;
 }
 
+const APPLAB_URL_PATTERN = /^(.*\/projects\/applab\/[A-Za-z0-9-]+)\/.*/;
+/**
+ * @returns the absolute url to the root of this project without a trailing slash.
+ *     For example: http://studio.code.org/projects/applab/GobB13Dy-g0oK
+ */
+function getProjectUrl() {
+  const match = location.href.match(APPLAB_URL_PATTERN);
+  if (match) {
+    return match[1];
+  }
+  return location.href; // i give up. Let's try this?
+}
+
```

Comment:
I'm not particularly happy/proud of this function. Seems like there should be a more centralized, uniform place for rendering URLs in javascript, but I'm trying to balance expediency with sensibility.